app/models/ingredient.rb

```diff
@@ -1,3 +1,10 @@
 class Ingredient < ApplicationRecord
    validates_presence_of :name, :cost
-end
\ No newline at end of file
+
+   has_many :recipe_ingredients
+   has_many :recipes, through: :recipe_ingredients
+
+   def self.alphabetical
+      @ordered_ingredients = Ingredient.order(:name)
```

Comment:
Typically a model method won't assign an instance variable to be returned, especially within a class method like this one, since the value can always be assigned to a different variable where this method is called. 

Having said that, your AR for this method is right on the money!

Reply:
I also don't see any model tests for this method - make sure your TDD process includes writing model methods first and that your coverage for models is always at 100%.